tests/integration/security/authorization_test.go

```diff
@@ -119,78 +115,80 @@ func TestAuthorization_JWT(t *testing.T) {
-					ExpectAllowed: expectAllowed,
-				}
-			}
-			cases := []rbacUtil.TestCase{
-				newTestCase(b, "[NoJWT]", "", "/token1", false),
-				newTestCase(b, "[NoJWT]", "", "/token2", false),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/token1", true),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/token2", false),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/token1", false),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/token2", true),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/tokenAny", true),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/tokenAny", true),
-				newTestCase(b, "[PermissionToken1]", jwt.TokenIssuer1, "/permission", false),
-				newTestCase(b, "[PermissionToken2]", jwt.TokenIssuer2, "/permission", false),
-				newTestCase(b, "[PermissionTokenWithSpaceDelimitedScope]", jwt.TokenIssuer2WithSpaceDelimitedScope, "/permission", true),
-				newTestCase(b, "[NoJWT]", "", "/tokenAny", false),
-				newTestCase(c, "[NoJWT]", "", "/somePath", true),
-
-				// Test condition "request.auth.principal" on path "/valid-jwt".
-				newTestCase(d, "[NoJWT]", "", "/valid-jwt", false),
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/valid-jwt", true),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/valid-jwt", true),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/valid-jwt", true),
-
-				// Test condition "request.auth.presenter" on suffix "/presenter".
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/request/presenter", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1, "/request/presenter", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/presenter-x", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/presenter", true),
-
-				// Test condition "request.auth.audiences" on suffix "/audiences".
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/request/audiences", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/audiences", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/request/audiences-x", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/request/audiences", true),
+			for _, srcCluster := range ctx.Clusters() {
+				ctx.NewSubTest(fmt.Sprintf("From a In %s", srcCluster.Name())).Run(func(ctx framework.TestContext) {
+					a := apps.A.Match(echo.InCluster(srcCluster).And(echo.NameSpace(ns.Name())))
+					callCount := 1
+					b := apps.B.Match(echo.NameSpace(ns.Name()))
+					c := apps.C.Match(echo.NameSpace(ns.Name()))
+					d := apps.D.Match(echo.NameSpace(ns.Name()))
+					if ctx.Clusters().IsMulticluster() {
+						// so we can validate all clusters are hit
+						callCount = util.CallsPerCluster * len(ctx.Clusters())
+					}
+					newTestCase := func(target echo.Instance, namePrefix string, jwt string, path string, expectAllowed bool, destClusters resource.Clusters) rbacUtil.TestCase {
```

Comment:
It seems if you just pass `destClusters` then you can save 1 argument from the caller? Inside newTestCase you can just use `destClusters.Default()` to get the target?

Reply:
Maybe I can use ```newTestCase := func(target echo.Instances, namePrefix string, jwt string, path string, expectAllowed bool)``` and use ```Request: connection.Checker{
								From: a[0],
								Options: echo.CallOptions{
									Target:   target[0],
									PortName: "http",
									Scheme:   scheme.HTTP,
									Path:     path,
									Count:    callCount,
								},
								DestClusters: target.Clusters(),
							},```. Is this OK?

Reply:
yeah, that also looks good, so you don't need to repeat the `.Clusters()` everywhere.